torch/_inductor/codegen/wrapper.py

```diff
@@ -1018,6 +1023,58 @@ def write_header(self):
                 """
             )
 
+    def write_inf_and_nan_header(self):
+        if self.inf_and_nan_header:
+            return
+
+        self.inf_and_nan_header = True
+        self.header.splice(
+            """
+            static inline bool check_has_inf(at::Tensor tensor, bool assertion) {
+              bool has_inf = false;
+
+              auto flattened = tensor.view({-1});
+              for (int64_t i = 0; i < flattened.numel(); i++) {
+                if(std::isinf(flattened[i].item<float>())) {
+                  has_inf = true;
+                  break;
+                }
+              }
+
+              if (assertion) {
+                assert(!has_inf);
```

Comment:
Is this `assert` macro from `<assert.h>`? Will it be optimized away with an `-O` flag in the compilation command?

Reply:
No, it won't be optimized away. In the test being ran locally, it do assert.